packages/widgetbook/lib/src/addons/grid_addon/grid_addon.dart

```diff
@@ -0,0 +1,105 @@
+import 'package:flutter/material.dart';
+
+import '../../fields/fields.dart';
+import '../common/common.dart';
+import 'grid_setting.dart';
+
+class GridAddon extends WidgetbookAddon<GridSetting> {
+  GridAddon({
+    int? initialHorizontalDistance,
+    int? initialVerticalDistance,
+  }) : super(
+          name: 'Grid',
+          initialSetting: GridSetting(
+            horizontalDistance: initialHorizontalDistance ?? 0,
+            verticalDistance: initialVerticalDistance ?? 0,
+          ),
+        );
+
+  @override
+  List<Field> get fields {
+    return [
+      IntField(
+        name: 'horizontalDistance',
+        initialValue: initialSetting.horizontalDistance,
+      ),
+      IntField(
+        name: 'verticalDistance',
+        initialValue: initialSetting.verticalDistance,
+      ),
+    ];
+  }
```

Comment:
I don't think that this Addon needs to be that much dynamic.
In other words we the can **remove the fields** completely and only **add 1 positional parameter called size** of type int, then users can use the addons as follows:

```dart
GridAddon(), // Default size could be 5 for example
GridAddon(4), // 4 will be used as both horizontalDistance & verticalDistance
```

Reply:
This has been attended to.